### PR TITLE
Examples: Ensure controls using the wheel event don't use document.

### PR DESCRIPTION
**Pull request by Mugen87**
see https://github.com/mrdoob/three.js/pull/16329#issuecomment-486802432

All controls utilizing the `wheel` event now log a warning if `document` is passed in as the second ctor parameter. Besides, this parameter (`domElement`) is now mandatory. Otherwise, the controls would use `document` as default value.

Should we make `domElement` to a mandatory parameter for all controls? It is already required for `DragControls` but not for `FlyControls`, `FirstPersonControls`, `TransformControls` and `PointerLockControls`.

### CODE REVIEW

**Comment by gkjohnson** [comment]
> Should we make domElement to a mandatory parameter for all controls?

I vote for yes. I think it would be great if all the controls classes were consistent in the way they handled events and use dom elements so it's easier to switch between them.

File: examples/js/controls/EditorControls.js [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 THREE.EditorControls = function ( object, domElement ) {
 
-	domElement = ( domElement !== undefined ) ? domElement : document;
+	if ( domElement === undefined ) console.error( 'THREE.EditorControls: The second paramter "domElement" is now mandatory. In most cases, "renderer.domElement" is best.' );
```
Comment (gkjohnson) [inline examples/js/controls/EditorControls.js]: > The second paramter "domElement"

"paramter" -> "parameter"

File: examples/js/controls/EditorControls.js [hunk 2]
```diff
@@ -7,7 +7,8 @@
...
+	if ( domElement === document ) console.warn( 'THREE.EditorControls: "document" is no valid value for "domElement". Please use "renderer.domElement" instead.' );
```
Comment (gkjohnson) [inline examples/js/controls/EditorControls.js]: > "document" is no valid value for "domElement"

Maybe instead:

"document" is not a valid value for "domElement"

or

"document" cannot be used as the target "domElement".

File: docs/examples/controls/OrbitControls.html [hunk 3]
```diff
@@ -56,8 +56,8 @@ <h3>[name]( [param:Camera object], [param:HTMLDOMElement domElement] )</h3>
 		<p>
 			[page:Camera object]: (required) The camera to be controlled.<br><br>
 
-			[page:HTMLDOMElement domElement]: (optional) The HTML element used for event listeners. By default this is the whole document,
-			however if you only want the controls to work over a specific element (e.g. the canvas) you can specify that here.
+			[page:HTMLDOMElement domElement]: The HTML element used for event listeners. In most cases, you want to pass in
+			the renderer's canvas element via *renderer.domElement*.
```
Comment (WestLangley) [inline docs/examples/controls/OrbitControls.html]: >In most cases, you want to pass in...

I do not think that is appropriate in documentation. If you want to go that route, you will have to say which cases.

File: examples/js/controls/OrbitControls.js [hunk 4]
```diff
@@ -17,7 +17,10 @@ THREE.OrbitControls = function ( object, domElement ) {
 
 	this.object = object;
 
-	this.domElement = ( domElement !== undefined ) ? domElement : document;
+	if ( domElement === undefined ) console.error( 'THREE.OrbitControls: The second parameter "domElement" is now mandatory. In most cases, "renderer.domElement" is best.' );
```
Comment (WestLangley) [inline examples/js/controls/OrbitControls.js]: >In most cases...

What I said above...

File: docs/examples/controls/OrbitControls.html (hunk 3 above)
Comment (Mugen87) [reply docs/examples/controls/OrbitControls.html]: I've removed this sentence.

**Comment by WestLangley** [comment]
Do you have to update the files in the `jsm` directory, too?

File: examples/js/controls/OrbitControls.js [hunk 5]
```diff
@@ -17,7 +17,10 @@ THREE.OrbitControls = function ( object, domElement ) {
 
 	this.object = object;
 
-	this.domElement = ( domElement !== undefined ) ? domElement : document;
+	if ( domElement === undefined ) console.error( 'THREE.OrbitControls: The second parameter "domElement" is now mandatory.' );
+	if ( domElement === document ) console.warn( 'THREE.OrbitControls: "document" cannot be used as the target "domElement". Please use "renderer.domElement" instead.' );
```
Comment (WestLangley) [inline examples/js/controls/OrbitControls.js]: Then checks for "document" elsewhere in the file need to be removed, right?

And this line should be a `console.error`, I guess.

**Comment by Mugen87** [comment]
I'm not sure. Is it sufficient to do this once right before a new release?

File: examples/js/controls/OrbitControls.js (hunk 5 above)
Comment (Mugen87) [reply examples/js/controls/OrbitControls.js]: Good point 👍

**Comment by WestLangley** [comment]
Perhaps you should refer to "the second argument" instead of the "the second parameter". Aren't they called constructor _arguments_?

**Comment by WestLangley** [comment]
>I'm not sure. Is it sufficient to do this once right before a new release?

Sorry, I don't understand your question.

**Comment by Mugen87** [comment]
I mean @mrdoob runs the modularize script before a new release, see https://github.com/mrdoob/three.js/commit/dafca9bed8a990166249aef3f12f8dcab80eaa66

Isn't this sufficient for `dev`?

**Comment by WestLangley** [comment]
I suppose updating jsm immediately would be helpful so we have a month to test it.

**Comment by Mugen87** [comment]
Done!

**Comment by looeee** [comment]
> Perhaps you should refer to "the second argument" instead of the "the second parameter". Aren't they called constructor arguments?

Here's what [Wikipedia says](https://en.m.wikipedia.org/wiki/Parameter_(computer_programming))

> if one defines the add subroutine as def add(x, y): return x + y, then x, y are parameters, while if this is called as add(2, 3), then 2, 3 are the arguments. 

Hmm, the terms are fairly interchangeable in this usage I'd say 🤔

**Comment by WestLangley** [comment]
>When dealing with methods, the term parameter is used to identify the placeholders in the method signature, whereas the term arguments are the actual values that you pass in to the method.

ref: https://stackoverflow.com/a/24367269/1461008

**Comment by EliasHasle** [comment]
Why was this closed? Forgetting to supply `domElement` is an ever recurring problem, ref. https://discourse.threejs.org/t/jsfiddle-templates-should-supply-renderer-domelement-to-orbitcontrols/9267

**Comment by mrdoob** [comment]
Sorry I didn't get to review this PR at the time. I vote for this too.
Do you want to give it another go?

**Comment by Mugen87** [comment]
Definitely 😍

**Pull request closed without merging**